pkg/koordlet/runtimehooks/hooks/cpuset/rule.go

```diff
@@ -56,27 +59,42 @@ func (r *cpusetRule) getContainerCPUSet(containerReq *protocol.ContainerRequest)
 		return nil, err
 	}
 
+	podQOSClass := ext.GetQoSClassByAttrs(podLabels, podAnnotations)
 	if len(podAlloc.NUMANodeResources) != 0 {
-		// LS pods which have specified cpu share pool
-		cpusetList := make([]string, 0, len(podAlloc.NUMANodeResources))
-		for _, numaNode := range podAlloc.NUMANodeResources {
-			for _, nodeSharePool := range r.sharePools {
-				if numaNode.Node == nodeSharePool.Node {
-					cpusetList = append(cpusetList, nodeSharePool.CPUSet)
+		if podQOSClass == ext.QoSBE && features.DefaultKoordletFeatureGate.Enabled(features.BECPUManager) {
+			// BE pods which have specified cpu share pool
+			cpusetList := make([]string, 0, len(podAlloc.NUMANodeResources))
+			for _, numaNode := range podAlloc.NUMANodeResources {
+				for _, nodeSharePool := range r.beSharePools {
+					if numaNode.Node == nodeSharePool.Node {
+						cpusetList = append(cpusetList, nodeSharePool.CPUSet)
+					}
+				}
+			}
```

Comment:
These codes are repeated with lines 79 - 85 (sharepool can be parameterized), and it is recommended to merge them.